### PR TITLE
Amount of creatures moved down by 1 pixel in DrawMonsterInfo

### LINKED ISSUE #1740: Castle screen, creature recruit window, creature information window with right click, the creature number "12" in not centered vertically and horizontally with respect to the creature number frame
In fheroes2, the creature number "12" in not centered vertically and horizontally with respect to the creature number frame.

In fheroes2:

Recruit creature info creature right fh2 01

### PR DESCRIPTION
**Pull request by shprotru**
Fix #1740

### CODE REVIEW

**Comment by ihhub** [comment]
Hi @shprotru , this is not a full fix for the issue. The biggest problem here is a horizontal position of numbers. Moreover, we have 2 types of window: good and evil interfaces and they have different area for such number.

This is a proper fix for only `good` interface:
```
pos.x = offset.x + 80 + 125 / 2 - text.w() / 2;
pos.y = offset.y + 222 + 23 / 2 - text.h() / 2 + 1;
```

but still we must avoid all these hardcoded values.

**Comment by idshibanov** [comment]
> but still we must avoid all these hardcoded values.

What's the approach for this you think? Most hardcoded offsets in UI I saw are related to original assets (shadows, GUI sprites) which won't change.

File: src/fheroes2/dialog/dialog_armyinfo.cpp [hunk 1]
```diff
@@ -401,10 +401,17 @@ void DrawMonsterInfo( const Point & offset, const Troop & troop )
...
+    const int offsetYGoodAmountBox = 222;
+    const int offsetXEvilAmountBox = 89;
+    const int offsetYEvilAmountBox = 221;
+    const int widthAmountBox = 125;
+    const int heightAmountBox = 23;
+
     text.Set( GetString( troop.GetCount() ), Font::BIG );
-    pos.x = offset.x + 140 - text.w() / 2;
-    pos.y = offset.y + 225;
+    pos.x = offset.x + ( Settings::Get().ExtGameEvilInterface() ? offsetXEvilAmountBox : offsetXGoodAmountBox ) + widthAmountBox / 2 - text.w() / 2;
```
Comment (ihhub) [inline src/fheroes2/dialog/dialog_armyinfo.cpp:413]: Please cache `Settings::Get().ExtGameEvilInterface()` variable: `const bool isEvilInterface = Settings::Get().ExtGameEvilInterface()`.

File: src/fheroes2/dialog/dialog_armyinfo.cpp [hunk 2]
```diff
@@ -401,10 +401,17 @@ void DrawMonsterInfo( const Point & offset, const Troop & troop )
     Point pos( offset.x + 140 - text.w() / 2, offset.y + 40 );
     text.Blit( pos );
 
-    // count
+    // amount
+    const int offsetXGoodAmountBox = 80;
```
Comment (ihhub) [inline src/fheroes2/dialog/dialog_armyinfo.cpp:405]: Put all these variables in unnamed namespace at the beginning of the file.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @shprotru , could you please take a look into 2 given comments?

File: src/fheroes2/dialog/dialog_armyinfo.cpp (hunk 2 above)
Comment (shprotru) [reply src/fheroes2/dialog/dialog_armyinfo.cpp]: done

File: src/fheroes2/dialog/dialog_armyinfo.cpp (hunk 1 above)
Comment (shprotru) [reply src/fheroes2/dialog/dialog_armyinfo.cpp]: done

**Comment by ihhub** [comment]
> > but still we must avoid all these hardcoded values.
> 
> What's the approach for this you think? Most hardcoded offsets in UI I saw are related to original assets (shadows, GUI sprites) which won't change.

At least we need to minimize these hardcoded values replacing them where needed by formulas. This will make our code more flexible for the future changes.

**Comment by ihhub** [comment]
@shprotru good job! Thank you very much for fixing this issue!

**Pull request merged**